Bug: ++/-- operators accept non-integer operands

## Description
The `++` and `--` postfix operators accept non-integer operands at compile time but fail at runtime.

## Reproduction
```ez
import @std
using std

do main() {
    temp s string = "hello"
    s++
    println(s)
}
```

Passes `ez check` but fails at runtime with:
```
error[E5023]: postfix operator ++ requires integer operand, got STRING
```

Also affects:
- `bool++` / `bool--`
- `float++` / `float--` (if not intentionally allowed)

## Expected Behavior
The type checker should validate that `++` and `--` are only used with integer types and produce a compile-time error for other types.

--- a/integration-tests/fail/errors/E3001_postfix_type_mismatch.ez
+++ b/integration-tests/fail/errors/E3001_postfix_type_mismatch.ez
@@ -1,0 +1,12 @@
+/*
+ * Error Test: E3001 - postfix-type-mismatch
+ * Expected: "postfix" and "integer"
+ */
+
+import @std
+using std
+
+do main() {
+    temp s string = "hello"
+    s++  // Should fail: increment on string
+}

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -1730,8 +1730,20 @@ func (tc *TypeChecker) checkRangeExpression(rangeExpr *ast.RangeExpression) {
 }
 
 // checkPostfixExpression validates postfix operators (++ and --)
-// These operators modify the operand, so we must check mutability
+// These operators modify the operand, so we must check mutability and type (#598)
 func (tc *TypeChecker) checkPostfixExpression(postfix *ast.PostfixExpression) {
+	// Check that operand is an integer type
+	operandType, ok := tc.inferExpressionType(postfix.Left)
+	if ok && !tc.isIntegerType(operandType) {
+		line, column := tc.getExpressionPosition(postfix.Left)
+		tc.addError(
+			errors.E3001,
+			fmt.Sprintf("postfix operator %s requires integer operand, got %s", postfix.Operator, operandType),
+			line,
+			column,
+		)
+	}
+
 	// Check mutability - error if trying to modify an immutable variable
 	if rootVar := tc.extractRootVariable(postfix.Left); rootVar != "" {
 		isMutable, found := tc.isVariableMutable(rootVar)

